src/test/java/com/exadel/movie_new/tests/MovieServiceTest.java

```diff
@@ -0,0 +1,158 @@
+        newMovie2.setGenres(List.of("Comedy", "Romance"));
+    }
+
+    @Test
+    @DisplayName("Should save the movie to a file")
+    void shouldAddNewMovie() {
+        doNothing().when(movieDaoImplMock).write(any());
+        Movie savedMovie = movieService.addMovie(newMovie);
+        assertNotNull(savedMovie);
+        assertFalse(savedMovie.getId().isEmpty());
+        assertEquals(newMovie.getTitle(), savedMovie.getTitle());
+        assertEquals(newMovie.getId(), savedMovie.getId());
+        assertEquals(newMovie.getGenres().get(0), savedMovie.getGenres().get(0));
+    }
+
+    @Test
+    @DisplayName("Should throw exception that movie already exists")
+    void shouldThrowExceptionMovieAlreadyExists() {
+        List<Movie> list = new ArrayList<>();
+        list.add(newMovie);
+        list.add(newMovie2);
+        when(movieDaoImplMock.read()).thenReturn(list);
+        assertThrows(MovieAlreadyExists.class, () -> movieService.addMovie(newMovie));
+    }
+
+    @Test
+    @DisplayName("Should fetch all movies")
+    void shouldGetAllMovies() {
+        List<Movie> list = new ArrayList<>();
+        list.add(newMovie);
+        list.add(newMovie2);
+        when(movieDaoImplMock.read()).thenReturn(list);
+        List<Movie> movies = movieService.getAllMovies();
+        assertNotNull(movies);
+        assertFalse(movies.isEmpty());
+        assertEquals(2, movies.size());
+    }
+
+    @Test
+    @DisplayName("Should return an empty list")
+    void shouldGetEmptyList() {
+        List<Movie> list = new ArrayList<>();
+        when(movieDaoImplMock.read()).thenReturn(list);
+        List<Movie> movies = movieService.getAllMovies();
+        assertNotNull(movies);
+        assertTrue(movies.isEmpty());
+        assertEquals(0,movies.size());
+    }
+
+    @Test
+    @DisplayName("Should fetch movie by Id")
+    void shouldGetMovieById() {
+        List<Movie> list = new ArrayList<>();
+        list.add(newMovie);
+
+        when(movieDaoImplMock.read()).thenReturn(list);
+        Movie existingMovie = movieService.getMovie(newMovie.getId());
+        assertNotNull(existingMovie);
+        assertNotEquals(null, existingMovie.getId());
+        assertEquals(newMovie.getId(), existingMovie.getId());
+    }
+
+    @Test
+    @DisplayName("Should throw MovieNotFoundException while getMovie")
+    void shouldGetMovieByIdException() {
+        List<Movie> list = new ArrayList<>();
+        list.add(newMovie);
+        when(movieDaoImplMock.read()).thenReturn(list);
+        assertThrows(MovieNotFoundException.class, () -> movieService.getMovie("200"));
+    }
+
+    @Test
+    @DisplayName("Should update movie")
+    void shouldUpdateMovie() {
+        List<Movie> list = new ArrayList<>();
+        list.add(newMovie);
+        when(movieDaoImplMock.read()).thenReturn(list);
+        doNothing().when(movieDaoImplMock).write(any());
+        newMovie.setTitle("Mad Max");
+        Movie existingMovie = movieService.updateMovie(newMovie);

```

Comment:
Same as with **addMovie** - you write movie into file but return same movie as passed into **updateMovie** without manipulation with it. In general such operations mist be **void** and only **get*** operations must return something.